flow/Net2.actor.cpp

```diff
@@ -604,9 +652,22 @@ class SSLConnection : public IConnection, ReferenceCounted<SSLConnection> {
 			int64_t permitNumber = wait(g_network->networkInfo.handshakeLock->take());
 			state BoundedFlowLock::Releaser releaser(g_network->networkInfo.handshakeLock, permitNumber);
 
-			BindPromise p("N2_AcceptHandshakeError", UID());
-			auto onHandshook = p.getFuture();
-			self->getSSLSocket().async_handshake( boost::asio::ssl::stream_base::server, std::move(p) );
+			Future<Void> onHandshook;
+
+			// If the background handshakers are not all busy, use one
+			if(N2::g_net2->sslPoolHandshakesInProgress < FLOW_KNOBS->TLS_HANDSHAKE_THREADS) {
+				usePool = true;
+				++N2::g_net2->sslPoolHandshakesInProgress;
+				auto handshake = new SSLHandshakerThread::Handshake(self->ssl_sock, boost::asio::ssl::stream_base::server);
+				onHandshook = handshake->done.getFuture();
+				N2::g_net2->sslHandshakerPool->post(handshake);
```

Comment:
Not a problem with this PR, but presumably `post` consumes `handshake`?  It'd be nice if this required a `std::move()` or some other indication that the compiler checked `handshake`'s lifecycle.  (This looks like a memory leak on first read.)

(The auto above resolves to a `PThreadAction`, which is a `ThreadAction *`, and eventually bottoms out in some boost  code with things like `BOOST_ASIO_MOVE_ARG` in it, so I assume it's consumed.  The documentation I found doesn't say...)

Reply:
The thread pool functions are not particularly well documented but yes post is consuming.  I followed other usage examples, which I suppose also look like memory leaks.